cmd/preview/list.go

```diff
@@ -72,16 +86,72 @@ func executeListPreviews(ctx context.Context, opts ListFlags) error {
 		}
 		return fmt.Errorf("failed to get preview environments: %s", err)
 	}
-	w := tabwriter.NewWriter(os.Stdout, 1, 1, 2, ' ', 0)
-	fmt.Fprintf(w, "Name\tScope\tSleeping\tLabels\n")
+	previewListOutput, err := getPreviewOutput(ctx, opts)
+	switch opts.output {
+	case "json":
+		if err != nil {
+			return err
+		}
```

Comment:
I think we should move `previewListOutput, err := getPreviewOutput(ctx, opts)` inside the switch, for both JSON and YAML.

When the user runs the "default" case, which is the most common, we don't need to process this output. It will be an extra line repeated but then each case is handled separately properly

Reply:
Isn't it a kind of little redundancy we're adding?

Reply:
> I don't think so, just both cases need that function call. What I do believe we should do is to generate an output only if it is strictly necessary. In the end we are executing a logic that in most cases is not going to be necessary as Andrea says. Of course we could refactor the code to call the function once for both cases (JSON and YAML) but I think in this case calling the function in both cases makes it pretty clear what logic to follow in both cases avoiding to generate the output in the default cases.

Sounds good. Thanks!